components/engine_rocks/src/raft_engine.rs

```diff
@@ -137,25 +148,26 @@ impl RaftEngine for RocksEngine {
     fn clean(
         &self,
         raft_group_id: u64,
-        state: &RaftLocalState,
+        local_state: &RaftLocalState,
+        apply_state: &RaftApplyState,
         batch: &mut Self::LogBatch,
     ) -> Result<()> {
         batch.delete(&keys::raft_state_key(raft_group_id))?;
-        let seek_key = keys::raft_log_key(raft_group_id, 0);
-        let prefix = keys::raft_log_prefix(raft_group_id);
-        if let Some((key, _)) = self.seek(&seek_key)? {
-            if !key.starts_with(&prefix) {
+        let first_index = if apply_state.has_truncated_state()
+            && apply_state.get_truncated_state().get_index() != 0
+        {
+            apply_state.get_truncated_state().get_index() + 1
```

Comment:
This is not trustworthy. When compacting logs, the truncated state is updated immediately, and a background job is started to do clean up. If TiKV is shutdown before the job finishes, there will be logs never be deleted until next log compact job.